chromium_src/net/http/transport_security_state.cc

```diff
@@ -130,6 +130,16 @@ GetHostBoundPartitionHashForHSTS(const std::string& host) {
 
 }  // namespace
 
+SSLUpgradeDecision TransportSecurityState::GetSSLUpgradeDecision(
```

Comment:
why are we doing things differently from upstream here? As with other places I have commented on we need the context for why we are making these changes and ideally it should be here in the code

Reply:
It's just a new entry point that needs the same change that was previously made here https://github.com/brave/brave-core/pull/13062 (Use schemeless key to partition HSTS data). I can add a comment explaining things.

Reply:
just some comments would be great

Reply:
Added additional comments in f4491d33a254e161ad0514958a8468068ab9c04d